code/modules/projectiles/updated_projectiles/ammo_datums.dm

```diff
@@ -1292,16 +1292,28 @@
 	damage_var_low = CONFIG_GET(number/combat_define/low_proj_variance)
 	damage_var_high = CONFIG_GET(number/combat_define/mlow_proj_variance)
 
+
 /datum/ammo/xeno/toxin/on_hit_mob(mob/living/carbon/M, obj/item/projectile/P)
+
 	if(!istype(M))
 		return ..()
 	var/mob/living/carbon/C = M
-	if(C.status_flags & XENO_HOST && istype(C.buckled, /obj/structure/bed/nest) || C.stat == DEAD)
+
+	if(C.status_flags & XENO_HOST && istype(C.buckled, /obj/structure/bed/nest) || C.stat == DEAD || !(ishuman(C) || ismonkey(C)))
 		return
+
 	staggerstun(C, P, CONFIG_GET(number/combat_define/close_shell_range), 0, 0, 1, 1, 0) //Staggers and slows down briefly
 
+	for(var/r_id in ammo_reagents)
+		var/on_mob_amount = C.reagents.get_reagent_amount(r_id)
+		var/amt_to_inject = ammo_reagents[r_id] //Never inject more than 30u through spitting
+		if(amt_to_inject + on_mob_amount > 30)
+			amt_to_inject = 30 - on_mob_amount
```

Comment:
OD still expires whenever the reagent amount goes below the threshold rather than on deletion. That said, this makes the OD simply last a lone tick at most, is this is your plan?

Reply:
Yep, I don't even see a need for xenos to ever OD marines on this.